### PR TITLE
Strike bugfix

### PR DESCRIPTION
**Pull request by lidio601**
https://github.com/lodash/lodash/commit/c6f8896826a82d894e0335e78e42d8d8f80bb87a

### CODE REVIEW

File: install [hunk 1]
```diff
@@ -160,7 +160,10 @@ cat <<EOF > $CONFIG_DIR/lamassu.json
   "lamassuServerPath": "$NODE_MODULES/lamassu-server",
   "migrateStatePath": "$MIGRATE_STATE_PATH",
   "blockchainDir": "$BLOCKCHAIN_DIR",
-  "ofacDataDir": "$OFAC_DATA_DIR"
+  "ofacDataDir": "$OFAC_DATA_DIR",
+  "strike": {
+    "baseUrl": null
```
Comment (joshmh) [inline install]: Put in the default correct URL: https://api.strike.acinq.co/api/

File: lib/plugins/layer2/strike/strike.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 const axios = require('axios')
-const _ = require('lodash/fp')
+const _ = require('lodash')
```
Comment (joshmh) [inline lib/plugins/layer2/strike/strike.js]: We always use `lodash/fp` as it allows us to be more functional. It's important to keep this consistent.

File: lib/plugins/layer2/strike/strike.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ module.exports = {
   cryptoNetwork
 }
 
-axios.defaults.baseURL = options.strike.baseUrl
+axios.defaults.baseURL = _.toString(_.get(options, 'strike.baseUrl', ''))
```
Comment (joshmh) [inline lib/plugins/layer2/strike/strike.js]: How does it help to default to blank baseURL? Should probably throw an error, instead.

File: lib/plugins/layer2/strike/strike.js [hunk 4]
```diff
@@ -38,15 +38,18 @@ function createCharge (account, info) {
   return axios({
     method: 'post',
     url: 'v1/charges',
-    auth: {username: account.token, password: ''},
+    auth: {
+      username: _.get(account, 'token', ''),
```
Comment (joshmh) [inline lib/plugins/layer2/strike/strike.js]: Token should never be blank. Throw error instead.

File: lib/plugins/layer2/strike/strike.js [hunk 5]
```diff
@@ -38,15 +38,18 @@ function createCharge (account, info) {
...
+    },
     data
   }).then(_.get('data'))
 }
 
 function newAddress (account, info) {
   return checkCryptoCode(info.cryptoCode)
     .then(() => createCharge(account, info))
-    .then(_.tap(console.log))
+//    .then(_.tap(console.log))
```
Comment (joshmh) [inline lib/plugins/layer2/strike/strike.js]: Just delete this line. Comments should only be descriptive text.

**Comment by lidio601** [comment]
this should be ready

File: lib/plugins/layer2/strike/strike.js [hunk 6]
```diff
@@ -9,7 +9,11 @@ module.exports = {
   cryptoNetwork
 }
 
-axios.defaults.baseURL = options.strike.baseUrl
+const baseUrl = _.attempt(() => options.strike.baseUrl)
```
Comment (joshmh) [inline lib/plugins/layer2/strike/strike.js]: Wouldn't `_.get` be simpler here?

File: lib/plugins/layer2/strike/strike.js [hunk 7]
```diff
@@ -34,19 +38,25 @@ function createCharge (account, info) {
     currency: 'btc',
     description: 'Lamassu cryptomat cash-out'
   }
+  const token = _.attempt(() => account.token)
```
Comment (joshmh) [inline lib/plugins/layer2/strike/strike.js]: Use `_.get`.

**Comment by lidio601** [comment]
updated

**Pull request merged**